server/members/views.py

```diff
@@ -224,3 +224,24 @@ class AdminList(generics.ListAPIView):
 
     def get_queryset(self):
         return Group.objects.get(name="is_admin").user_set.all()
+
+
+class UpdateDiscordUsername(APIView):
+    permission_classes = [~IsVerified]
+
+    def post(self, request):
+        new_discord_username = request.data.get("new_discord_username")
+        member_id = request.data.get("member_id")
```

Comment:
I do'nt think we should be passing in member id, you should be using `request.user`. With your current code anyone can passi n a member_id and change someone's discord if I'm remembering correctly.

Reply:
It will also return the user so you don't haveto search. Then you can just update the users discord username.